swap/src/main.rs

```diff
@@ -334,6 +365,7 @@ async fn init_wallets(
     bitcoin_network: bitcoin::Network,
     bitcoin_wallet_data_dir: &Path,
     monero_network: monero::Network,
+    seed: Seed,
```

Comment:
Isn't the `seed` loaded from the `config_path`? We are we passing it in separately?

Reply:
The seed is stored in the data-dir, that can be configured with `--data-dir` and is different than `--config` which specifies the path to the configuration file. Note that on Linux the default paths are also not the same :)

Additionally, we use the seed for two things, for deriving the peer-id and the wallet private key. The seed is read at the very top of main.

Reply:
It is still inconsistent then. I would expect all handling of config / data dir _paths_ to be on the same abstraction level. If this component receives the _config_ path as a parameter, it should also take the _data-dir_ path as a parameter.

Alternatively, if you are reading the seed outside and passing it in, I would also expect whatever the function does with the config path to do that outside and pass it in.

Reply:
I totally agree. We should not pass a config path here, but should the the config!
Loading the config inside `init_wallets` was a mistake (that I introduced...) because at the moment only the wallet URLs are stored in the config. So I handled the initial setup (as in querying the user for said URLs) in `init_wallets`. 
But that is generally not a good abstraction. It causes troubles with being able to create only one wallet for specific commands as well. I tried to sum this up in https://github.com/comit-network/xmr-btc-swap/issues/181

Reply:
Thank you for elaborating and creating the ticket!